sql/expression/convert.go

```diff
@@ -0,0 +1,212 @@
+package expression
+
+import (
+	"encoding/json"
+	"fmt"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/spf13/cast"
+	errors "gopkg.in/src-d/go-errors.v0"
+	"gopkg.in/src-d/go-mysql-server.v0/sql"
+)
+
+// ErrConvertExpression is returned when a conversion is not possible.
+var ErrConvertExpression = errors.NewKind("expression '%v': couldn't convert to %v")
+
+const (
+	// ConvertTo represents those conversion types Convert accepts.
+	ConvertToBinary   = "binary"
```

Comment:
are these constants used outside this package? If not, I'd keep them lowercase. And if so, I would make them integers, the take less space and the string is not really needed. They're only needed as an enum

Reply:
Those are the string go-vitess' parser returns, I can change it to enums, but I'd have to create two maps kept in memory to translate strings->int and int->strings and make the logic a little bit complex.

They are only used inside the expression package, but I think it's ok to have them public if at some point we need to create `Convert` expressions in a rule or something like that. 

Do you think it will make a really improve the enum thing?

Reply:
Then I think the enum is not worth it, too much hassle for so little benefit. LGTM as is, then.